### PR TITLE
Mc celery

### PR DESCRIPTION
**Pull request by FBRTMaka**
@DanielJMaher @oceanzus ,

Here's the work for celery - it's SUPER picky, so some things may look strange but it's how it goes.

Another process will need to be kicked off while in the virtualenv in the ooi-ui-services root dir after redis is running:

celery worker -A ooiservices.app.tasks -B

### CODE REVIEW

**Comment by DanielJMaher** [comment]
@oceanzus This will need to be worked into your deployment strategy, so I'm not sure what you need to do to this and with this, but have at it good sir.

File: ooiservices/app/__init__.py [hunk 1]
```diff
@@ -99,9 +99,29 @@ def create_app(config_name):
...
-
     return app
 
+
+from celery import Celery
+
+
+def create_celery_app(env, app=None):
+    app = app or create_app(env)
+    celery = Celery('__main__', broker='redis://localhost:6379/0')
```
Comment (oceanzus) [inline ooiservices/app/__init__.py]: This needs to come from the config file.

File: ooiservices/app/tasks.py [hunk 2]
```diff
@@ -1,51 +1,87 @@
...
 '''
 Create the celery app, and configure it to talk to the redis broker.
 Then intialize it.
 '''
-celery_app = Flask(__name__)
-
-# Celery configuration
-celery_app.config['CELERY_BROKER_URL'] = os.environ.get('REDISCLOUD_URL') or \
-    'redis://localhost:6379/0'
-celery_app.config['CELERY_RESULT_BACKEND'] = os.environ.get('REDISCLOUD_URL') or \
```
Comment (oceanzus) [inline ooiservices/app/tasks.py]: Needs to come from the config file

**Comment by oceanzus** [comment]
Why is manage.py changed in this PR?

File: ooiservices/app/tasks.py (hunk 2 above)
Comment (FBRTMaka) [inline ooiservices/app/tasks.py]: This is a deletion lol

File: ooiservices/app/__init__.py (hunk 1 above)
Comment (FBRTMaka) [inline ooiservices/app/__init__.py]: It can't, because the app hasn't been created yet.

File: ooiservices/manage.py [hunk 3]
```diff
@@ -178,11 +188,28 @@ def deploy(password, bulkload, production, psqluser):
...
+        with open('db/ooiui_params_streams_data.sql') as ps:
+            psql('-U', psqluser, 'ooiuiprod', _in=ps)
+
+        app.logger.info('Populating Dev Database . . .')
+        with open('db/ooiui_schema_data.sql') as dd:
+            psql('-U', psqluser, 'ooiuidev', _in=dd)
+
+        with open('db/ooiui_params_streams_data.sql') as ds:
+            psql('-U', psqluser, 'ooiuidev', _in=ds)
+
```
Comment (DanielJMaher) [inline ooiservices/manage.py]: Was the inclusion of this accidental?  It doesn't seem like it is related to celery.

Comment (DanielJMaher) [inline ooiservices/manage.py]: This needs to be evaluated by Jim.

Comment (FBRTMaka) [inline ooiservices/manage.py]: @DanielJMaher Well yeah! That's why I added him to the review list.

@oceanzus This SHOULD make it using the 'deploy' param better to prod while at the same time not impacting what ever custom deployment plan you have going on.

Comment (oceanzus) [inline ooiservices/manage.py]: @FBRTMaka https://github.com/asascience-open/ooi-ui-services/pull/447

Comment (FBRTMaka) [inline ooiservices/manage.py]: @oceanzus sweet! I just updated manage.py to reflect #447

**Comment by oceanzus** [comment]
That is an improper merge.  I suggest deleting this PR and just using mine, which was based on this PR and maintains all the Celery code.

**Comment by FBRTMaka** [comment]
@oceanzus damn Jim, not sure what the fuck you did that manage.py but it's working way outside of context.  reverting and forcing the correct code back.

**Comment by FBRTMaka** [comment]
screen shot 2015-11-19 at 10 11 38 pm

**Comment by FBRTMaka** [comment]
@DanielJMaher @oceanzus I suggest you don't try to be cute with this celery stuff, it's a world of shit for the uninitiated.

**Comment by DanielJMaher** [comment]
@FBRTMaka @oceanzus I'm making the call here.  We are moving forward with Jim's .  Matt, if you have problems with his PR we can address them there.

**Comment by FBRTMaka** [comment]
Roger that, using Jim’s.

On Nov 20, 2015, at 9:00 AM, Dan Maher <notifications@github.com<mailto:notifications@github.com>> wrote:


@FBRTMaka<https://github.com/FBRTMaka> @oceanzus<https://github.com/oceanzus> I'm making the call here. We are moving forward with Jim's . Matt, if you have problems with his PR we can address them there.

—
Reply to this email directly or view it on GitHub<https://github.com/asascience-open/ooi-ui-services/pull/446#issuecomment-158409561>.


This e-mail message and any attached file is the property of the sender and is sent in confidence to the addressee only.
Internet communications are not secure and RPS is not responsible for their abuse by third parties, any alteration or corruption in transmission or for any loss or damage caused by a virus or by other means.
 
Any advice contained in this e-mail is for information purposes only.

RPS Group Plc, company number: 208 7786 (England). Registered office: 20 Western Avenue Milton Park Abingdon Oxfordshire OX14 4SH.
RPS Group Plc web link: <http://www.rpsgroup.com>

**Pull request closed without merging**